packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -180,14 +208,99 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     });
   }
 
+  // Get indices of selected elements in options array when we have selected values.
+  const getSelectedIndexes = React.useCallback(() => {
+    const indexArray: number[] = [];
+    if (isMultipleEnabled(valueProp, multiple) && !!valueProp) {
+      valueProp.forEach((value) => {
+        indexArray.push(options.findIndex((option) => option.value === value));
+      });
+    }
+    return indexArray;
+  }, [multiple, options, valueProp]);
+
   // Reducer where all the component-wide state is stored
+  const comboBoxReducer = (
+    state: {
+      isOpen: boolean;
+      selectedIndex: number | number[];
```

Comment:
`selectedIndex` implies that it's a number, not an array. Maybe rename to `selected`.

Although, it might also be better to have separate properties `selectedIndex` and `selectedIndices` (both optional) and use them depending on whether `multiple` is false/true.

Reply:
Renamed to selected. Not sure about having two properties.